scikeras/wrappers.py

```diff
@@ -64,10 +67,71 @@ class BaseWrapper(BaseEstimator):
         "multioutput": True,
     }
 
+    _fit_params = {
+        # parameters destined to keras.Model.fit
+        "callbacks",
+        "batch_size",
+        "epochs",
+        "verbose",
+        "callbacks",
+        "validation_split",
+        "shuffle",
+        "class_weight",
+        "sample_weight",
+        "initial_epoch",
+        "validation_steps",
+        "validation_batch_size",
+        "validation_freq",
+    }
+
+    _predict_params = {
+        # parameters destined to keras.Model.predict
+        "batch_size",
+        "verbose",
+        "callbacks",
+        "steps",
+    }
+
+    _compile_params = {
+        # parameters destined to keras.Model.compile
+        "optimizer",
+        "loss",
+        "metrics",
+        "loss_weights",
+        "weighted_metrics",
+        "run_eagerly",
+    }
+
+    _wrapper_params = {
+        # parameters consumed by the wrappers themselves
+        "warm_start",
+        "random_state",
+    }
+
+    _meta_params = {
+        # parameters created by wrappers within `fit`
+        "_random_state",
+        "n_features_in_",
+        "X_dtype_",
+        "y_dtype_",
+        "X_shape_",
+        "y_shape_",
+        "model_",
+        "history_",
+        "is_fitted_",
+        "n_outputs_",
+        "keras_expected_n_ouputs_",
```

Comment:
Although it is not determined by the Keras Model, it is checked against it: https://github.com/adriangb/scikeras/blob/250837efa4476b2647a6aaf9381306070485469c/scikeras/wrappers.py#L299

Reply:
I'm not a fan of the name `keras_expected_n_outputs_` because I can't tell how it's computed or where it comes from.

I think better names would be `model_n_outputs_` or `target_type_n_outputs_`. I have a better idea where those come from: from the model and data respectively.

Reply:
I like `model_n_outputs_`, happy to change it to that

Reply:
Renamed to `model_n_outputs_`.